### DISCUSSION TITLE
Error: unexpected error encountered: API rate limit exceeded for installation ID 12770951.

### DISCUSSION THREAD

**Discussion by gajus** [bobheadxi/deployments#81] [Help]
```
Run bobheadxi/deployments@v0.6.1
  with:
    deployment_id: 469535842
    env_url: https://***/
    status: success
    step: finish
    token: ***
    no_override: true
    log_args: false
targeting contra/contra-web-app
finishing deployment for 469535842 with status success
Error: unexpected error encountered: API rate limit exceeded for installation ID 12770951.
```

Not clear where this error is coming from or how to mitigate it.

**Comment by bobheadxi** [comment]
It appears you are hitting the rate limit for your GitHub credentials or application - what token is provided? Is it from a bot account that is used for a lot of automation on this repository?

**Comment by bobheadxi** [comment]
@gajus I'm pretty sure this can be fixed by creating a bot account, and using a token from a bot account to ensure that the bot will always have available API request quotas :) Please feel free to ping me if you have any more questions!

(also very cool to see Contra using this action! 😁 )

**Comment by NickGraham101** [comment]
We encountered a similar issue.  In our case we were using the action to update a deployment in an environment that already had a large number of deployments (over 100).  When the action looped through each previous deployment setting it to inactive we would eventually get a rate limiting error after about 50-60 deployments had been updated.

In the logs we'd see:

```
sandbox: found 100 existing deployments for env
sandbox.525806776: setting deployment (c6545a428a2c2713265353370443419716c3154c) state to "inactive"
sandbox.525784693: setting deployment (cacf304bc9725d2620582d2b9d2818fe8bd6973e) state to "inactive"
sandbox.525193098: setting deployment (ecf25276076bb5051398d99a8a52135d01282bfc) state to "inactive"
... (~50 similar lines omitted)
sandbox.509878615: setting deployment (6abd0e6e535a63736e7c0f8e202f2271ceb31924) state to "inactive"
unexpected error encountered: HttpError: API rate limit exceeded for installation ID <redacted>.
```

GitHub [allows](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits) 1000 requests per hour, so for repos with low usage this isn't a problem.  However in our case we're deploying to multiple environments many times an hour during busy periods so it's quite easy to hit the 1000 request limit when each deployment can result in 100+ calls to the API to update the deployment status.

**Reply by bobheadxi** [reply]
Hey everyone, unfortunately, I haven't been able to take a look at this and probably won't have the bandwidth to actively maintain this project for the time being. If someone finds a good way forward I would appreciate a PR! I just cut https://github.com/bobheadxi/deployments/releases/tag/v1.2.0 which includes a recent contribution which changes up how delete-env works a bit, though not in a way that resolves the issue here I think: https://github.com/bobheadxi/deployments/pull/101

**Reply by bobheadxi** [reply]
If anyone is open to helping me maintain this project I'd be super grateful and happy to chat, please reach out to me via [twitter](https://twitter.com/bobheadxi), email, or [this discussion](https://github.com/bobheadxi/deployments/discussions/103)! 🙏
